src/bgw/job_stat.c

```diff
@@ -198,8 +199,25 @@ calculate_next_start_on_failure(TimestampTz finish_time, int consecutive_failure
 	TimestampTz res = 0;
 	volatile bool res_set = false;
 	TimestampTz last_finish = finish_time;
-	float8 multiplier = (consecutive_failures > MAX_FAILURES_MULTIPLIER ? MAX_FAILURES_MULTIPLIER :
-																		  consecutive_failures);
+	int exponent = (consecutive_failures > MAX_FAILURES_MULTIPLIER ? MAX_FAILURES_MULTIPLIER :
+																	 consecutive_failures);
+	/*
+	 * the aim of backoff is to reduce conflicts in the case of jobs that fail to launch,
+	 * as there are jobs competing for available background workers in that case.
+	 * therefore we introduce some randomness in backoff calculation for that type of failure
+	 * by randomly selecting a backoff time in the interval [2, 2^consecutive_failures]
+	 * Otherwise as soon as a worker becomes available, all jobs will try to grab it, but only one
+	 * will anyway so why should they all try at once? so jobs that experience failures will delay
+	 * their next_start a random amount to reduce the probability of collisions
+	 */
+	Assert(consecutive_failures > 0);
+	// 2^(consecutive_failures) - 1, at most 2^20
+	int64 max_slots = (1 << consecutive_failures) - 1;
+	// granularity: 1 usec (2^20)
+	int64 micros_per_minute = 1000000;
+	// will get a random int in [0, (2^f - 1) * 1000]
+	// this represents a random amount of microseconds to backoff
+	int64 rand_backoff = random() % (max_slots * micros_per_minute);
```

Comment:
```suggestion
	int64 rand_backoff = random() % (max_slots * USECS_PER_SEC);
```
I don't see anything to do with minutes here.

Reply:
Indeed